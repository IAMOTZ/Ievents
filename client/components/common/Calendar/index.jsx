import React from 'react';
import PropTypes from 'prop-types';
import InfiniteCalendar from 'react-infinite-calendar';
import 'react-infinite-calendar/styles.css';

const Calendar = (props) => {
  const today = new Date();
  const tommorrow = new Date(today.getFullYear(), today.getMonth(), today.getDate() + 1);
  let minDate;
  // There must be 24hours between today and the minimum date that is selectable.
  if (tommorrow.getTime() - today.getTime() >= 24 * 60 * 60 * 1000) {
    minDate = tommorrow;
  } else {
    minDate = new Date(today.getFullYear(), today.getMonth(), today.getDate() + 2);
  }
  const maxMonth = new Date(today.getFullYear(), today.getMonth() + 12, today.getDate());
  const minMonth = new Date(today.getFullYear(), today.getMonth(), today.getDate());
  const selectedDate = props.selectedDate ? new Date(props.selectedDate) : false;

  return (
    <InfiniteCalendar
      width={400}
      height={300}
      selected={selectedDate}
      minDate={minDate}
      max={maxMonth}
      min={minMonth}
      onSelect={props.handleDateSelection}
      displayOptions={{
        showTodayHelper: false
      }}
      theme={{
        selectionColor: 'rgb(0, 142, 214)',
      }}
    />
  );
};

Calendar.defaultProps = {
  selectedDate: null,
};

Calendar.propTypes = {
  handleDateSelection: PropTypes.func.isRequired,
  selectedDate: PropTypes.string,
};

export default Calendar;
