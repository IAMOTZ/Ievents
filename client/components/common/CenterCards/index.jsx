import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';

const defaultImage = '/images/defaultImgx4.jpeg';

const CenterCards = props => props.centers.map(center => (
  <div className="col-lg-4 col-md-6 col-sm-1" key={center.id}>
    <div className="card centerCard">
      <div>
        <img className="card-img-top" src={center.images ? center.images[0] : defaultImage} alt="The Center" />
        <div className="badge-container">
          <span className="badge text-white p-2 seat-badge ">{center.capacity} seats</span>
        </div>
      </div>
      <div className="card-body d-flex flex-column">
        <div>
          <h4 className="card-title text-capitalize">{center.name}</h4>
          <div className="icons">
            <div>
              <i className="fa fa-map-marker fa-fw" aria-hidden="true" />&nbsp;
              <span className="text-capitalize">{center.location}</span>
            </div>
            <div>
              <i className="fa fa-info-circle fa-fw" aria-hidden="true" />&nbsp;
              <a data-toggle="collapse" href={`#center-details-collapse-${center.id}`} role="button">About center</a>
            </div>
          </div>
          <div className="collapse" id={`center-details-collapse-${center.id}`}>
            <br />
            <span className="center-details">{center.details}</span>
          </div>
        </div>
        {
          props.isAdmin ?
            <div className="admin-only-icons">
              <i className="fa fa-pencil fa-fw" aria-hidden="true" />&nbsp;
              <Link
                to="/editCenter"
                id={center.id}
                onClick={props.onEdit}
              >Edit Center
              </Link>
            </div> : null
        }
        <Link
          to="/addEvent"
          onClick={props.onBook}
          id={center.id}
          className="btn ie-blue-button mt-2 btn-block"
        > Book now for â‚¦{center.price}/Day
        </Link>
      </div>
    </div>
  </div>
));

/* eslint-disable react/forbid-prop-types */
CenterCards.propTypes = {
  centers: PropTypes.array.isRequired,
  onBook: PropTypes.func.isRequired,
  onEdit: PropTypes.func,
};

export default CenterCards;
