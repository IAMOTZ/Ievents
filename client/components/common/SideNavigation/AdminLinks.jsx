import React from 'react';
import { Link } from 'react-router-dom';
import FakeDiv from '../../hoc/FakeDiv';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <FakeDiv>
        <Link to="/addCenter" className="list-group-item">
          <i className="fa fa-plus fa-fw" aria-hidden="true" />&nbsp; Add Center
        </Link>
        {
          props.isSuperAdmin ?
            <Link to="/addAdmin" className="list-group-item">
              <i className="fa fa-user fa-fw" aria-hidden="true" />&nbsp; Add Admin
            </Link> : null
        }
        <Link to="/transactions" className="list-group-item">
          <i className="fa fa-tasks fa-fw" aria-hidden="true" />&nbsp; Transactions
        </Link>
      </FakeDiv>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
