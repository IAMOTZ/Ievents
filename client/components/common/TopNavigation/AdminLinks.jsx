import React from 'react';
import { Link } from 'react-router-dom';
import FakeDiv from '../../hoc/Aux';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <FakeDiv>
        <li className="nav-item">
          <Link to="/addCenter" className="nav-link">Add Center</Link>
        </li>
        {
          props.isSuperAdmin ?
            <li className="nav-item">
              <Link to="/addAdmin" className="nav-link">Add Admin</Link>
            </li> : null
        }
        <li className="nav-item">
          <Link to="/transactions" className="nav-link">Transactions</Link>
        </li>
      </FakeDiv>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
