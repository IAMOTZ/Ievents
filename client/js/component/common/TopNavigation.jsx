import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { clearUser } from '../../actions/authAction';

@connect((store) => {
  return {

  }
})

// The navigation that is specific to user page
class UserTopNav extends React.Component {
  logout = () => {
    this.props.dispatch(clearUser());
  }
  render() {
    return (
      <div>
        <nav class="navbar navbar-expand-sm navbar-dark fixed-top d-block d-lg-none">
          <div class="container">
            <a href="" class="navbar-brand">
              <strong class="text-capitalize">{this.props.name}</strong>
              <br />
              <p class="h6">{this.props.title}</p>
            </a>
            <button class="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
              <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="navMenu">
              <ul class="navbar-nav ml-auto">
                <li class="nav-item">
                  <Link to="/events" className="nav-link">My events</Link>
                </li>
                <li class="nav-item">
                  <Link to="/addEvent" className="nav-link">Add Events</Link>
                </li>
                <li class="nav-item">
                  <Link to="#" className="nav-link">Centers</Link>
                </li>
                <li class="nav-item">
                  <Link to="/" className="nav-link" onClick={this.logout}>Logout</Link>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    );
  }
}

// The navigation that is returned by default
export default () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark fixed-top">
      <div className="container">
        <Link to="/" className="navbar-brand">
          <strong>Ievents</strong>
        </Link>
        <button className="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navMenu">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link to="/" className="nav-link">Home</Link>
            </li>
            <li className="nav-item">
              <Link to="#" className="nav-link">Centers</Link>
            </li>
            <li className="nav-item">
              <Link to="/users/login" className="nav-link"> Signin </Link>
            </li>
            <li className="nav-item">
              <Link to="/users" className="nav-link"> Signup </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}

// The navigation that is specific to admin page  
const AdminTopNav = (props) => {
  return (
    <div>
      <nav class="navbar navbar-expand-sm navbar-dark fixed-top d-block d-lg-none">
        <div class="container">
          <a href="" class="navbar-brand">
            <strong>Admin</strong>
            <br />
            <p class="h6">{props.title}</p>
          </a>
          <button class="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
            <span class="navbar-toggler-icon"></span>
          </button>
          <div class="collapse navbar-collapse" id="navMenu">
            <ul class="navbar-nav ml-auto">
              <li class="nav-item">
                <Link to="#" class="nav-link">Centers</Link>
              </li>
              <li class="nav-item">
                <Link to="#" class="nav-link">Add Center</Link>
              </li>
              <li class="nav-item">
                <Link to="#" class="nav-link">Transactions</Link>
              </li>
              <li class="nav-item">
                <Link to="#" class="nav-link">Logout</Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
};

export { UserTopNav, AdminTopNav };
