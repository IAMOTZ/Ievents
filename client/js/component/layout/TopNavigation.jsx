import React from 'react';

export default () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark fixed-top">
      <div className="container">
        <a href="index.html" className="navbar-brand">
          <strong>Ievents</strong>
        </a>
        <button className="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navMenu">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <a href="index.html" className="nav-link">Home</a>
            </li>
            <li className="nav-item">
              <a href="centers.html" className="nav-link">Centers</a>
            </li>
            <li className="nav-item">
              <a href="signin.html" className="nav-link">Signin</a>
            </li>
            <li className="nav-item">
              <a href="#" className="nav-link">Signup</a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
