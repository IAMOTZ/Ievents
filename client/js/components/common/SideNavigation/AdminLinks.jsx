import React from 'react';
import { Link } from 'react-router-dom';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <div>
        <Link to="/addCenter" className="list-group-item">
          <i className="fa fa-plus fa-fw" aria-hidden="true" />&nbsp; Add Center
        </Link>
        {
          props.isSuperAdmin ?
            <Link to="/addAdmin" className="list-group-item">
              <i className="fa fa-user fa-fw" aria-hidden="true" />&nbsp; Add Admin
            </Link> : null
        }
        <Link to="/transactions" className="list-group-item">
          <i className="fa fa-tasks fa-fw" aria-hidden="true" />&nbsp; Transactions
        </Link>
      </div>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
