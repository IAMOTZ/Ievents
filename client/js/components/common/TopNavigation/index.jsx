import RegularTopNavigation from './RegularTopNavigation';
import AuthTopNavigation from './AuthTopNavigation';

export { RegularTopNavigation, AuthTopNavigation };
