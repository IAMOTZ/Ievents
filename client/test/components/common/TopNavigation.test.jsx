import React from 'react';
import { RegularTopNavigation, AuthTopNavigation } from '../../../components/common/TopNavigation';
import AdminLinks from '../../../components/common/TopNavigation/AdminLinks';

const props = {
  name: 'Tester',
  title: 'Testing',
  isAdmin: false,
  isSuperAdmin: false,
  dispatch: jest.fn(),
};

const alterProps = newProps => ({ ...props, ...newProps });

describe('<RegularTopNavigation />', () => {
  describe('Rendering:', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<RegularTopNavigation />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});


describe('<AuthTopNavigation />', () => {
  describe('Rendering:', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<AuthTopNavigation {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Interaction', () => {
    it('should dispatch the logout action', () => {
      const wrapper = shallow(<AuthTopNavigation {...props} />);
      wrapper.find('Link[to="/"]').simulate('click');
      expect(props.dispatch.mock.calls.length).toEqual(1);
    });
  });
});

describe('<AdminLinks />', () => {
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AdminLinks {...props} />);
    });
    it('should render correctly if the user is not an admin', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly if the user is an admin', () => {
      wrapper.setProps(alterProps({ isAdmin: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly if the user is a super admin', () => {
      wrapper.setProps(alterProps({ isSuperAdmin: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
