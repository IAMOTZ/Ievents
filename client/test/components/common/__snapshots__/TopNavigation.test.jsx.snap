// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLinks /> Rendering should render correctly if the user is a super admin 1`] = `
<Aux>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/addCenter"
    >
      Add Center
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/addAdmin"
    >
      Add Admin
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/centers/transactions"
    >
      Transactions
    </Link>
  </li>
</Aux>
`;

exports[`<AdminLinks /> Rendering should render correctly if the user is an admin 1`] = `
<Aux>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/addCenter"
    >
      Add Center
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/centers/transactions"
    >
      Transactions
    </Link>
  </li>
</Aux>
`;

exports[`<AdminLinks /> Rendering should render correctly if the user is not an admin 1`] = `""`;

exports[`<AuthTopNavigation /> Rendering: should render correctly 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark fixed-top d-block d-lg-none"
  id="auth-top-navbar"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <strong
        className="text-capitalize"
      >
        Tester
      </strong>
      <br />
      <p
        className="h6"
      >
        Testing
      </p>
    </a>
    <button
      className="navbar-toggler"
      data-target="#navMenu"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navMenu"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/events"
          >
            My events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/centers"
          >
            Centers
          </Link>
        </li>
        <AdminLinks
          isAdmin={false}
          isSuperAdmin={false}
        />
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/profile"
          >
            Profile
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`<RegularTopNavigation /> Rendering: should render correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark fixed-top"
  id="regular-top-navbar"
>
  <div
    className="container"
  >
    <Link
      className="navbar-brand"
      replace={false}
      to="/"
    >
      <strong>
        iEvents
      </strong>
    </Link>
    <button
      className="navbar-toggler"
      data-target="#navMenu"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navMenu"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/explore/centers"
          >
            Centers
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/users/login"
          >
             Log in 
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/users/signup"
          >
             Sign up 
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
